src/components/App.tsx

```diff
@@ -317,6 +319,24 @@ export class App extends React.Component<AppProps, AppState> {
     const projectModel = this.state.project.getModel();
     this.controlCenter.sandbox.run(projectModel, src);
   }
+
+  showPreview(file: File) {
+    if (file.type === FileType.Markdown) {
+      updateFilePreview(file);
+      const { preview } = file;
+
+      const onBufferChange = () => updateFilePreview(file);
+      file.onDidChangeBuffer.register(onBufferChange);
+      preview.onClose = () => {
+        file.onDidChangeBuffer.unregister(onBufferChange);
+        removeFilePreview(file);
+      };
```

Comment:
can this move into updateFilePreview? btw, the way the action perform the better name for it will be `addFilePreview`

Reply:
`showPreview` is called once the preview button is clicked. So here we set up the listeners for change in the original markdown file.

We can move it there but should we?

Reply:
If we move then `updateFilePreview` action will be doing a lot of things

Reply:
> If we move then updateFilePreview action will be doing a lot of things

that's the point :) App.tsx shall not do anything